src/pynxtools_raman_multiformat/reader.py

```diff
@@ -0,0 +1,242 @@
+            line_count += 1
+            line = line.strip()  # Remove any leading/trailing whitespace
+            if line.startswith("[Header]"):
+                current_section = "header"
+                continue
+            elif line.startswith("[Data]"):
+                header_length = line_count + 2
+                current_section = "data"
+
+                continue
+
+            # Parse the header section
+            if current_section == "header" and "=" in line:
+                key, value = line.split("=", 1)
+                header_dict[key.strip()] = value.strip()
+
+            # Parse the data section
+            elif current_section == "data" and "," in line:
+                if line_count <= header_length:
+                    if line.startswith("[Header]"):
+                        print("We have to do something here")
+                if line_count > header_length:
+                    values = line.split(",")
+                    data.append([float(values[0].strip()), float(values[1].strip())])
+
+        #transform linewise read data to colum style data
+
+
+
+        # Transform: [[A, B], [C, D], [E, F]] into [[A, C, E], [B, D, F]]
+        data = [list(item) for item in zip(*data)]
+
+        data = np.transpose(data)
+
+        data_dict = {
+            "data/x_values": data[:, 0],
+            "data/y_values": data[:, 1]
+        }
+
+        self.txt_data = data_dict
+        self.txt_header = header_dict
+
+
+    def read_txt_columns(self, filepath):
+        data = np.loadtxt(filepath, delimiter=',')
+
+        # Convert the array into a dictionary with labeled columns
+        data_dict = {
+            "data/x_values": data[:, 0],
+            "data/y_values": data[:, 1]
+        }
+
+        A=self.transform_nm_to_wavenumber(532.1,data_dict["data/x_values"])
+        print(A,"A")
+        return data_dict
+
+    def get_eln_data(self, key: str, path: str) -> Any:
+        """Returns data from the given eln path."""
+        if self.eln_data is None:
+            return None
+        upper_and_lower_mixed_nexus_concepts = ["/detector_TYPE[",
+                                        "/beam_TYPE[",
+                                        "/source_TYPE[",
+                                        "/polfilter_TYPE[",
+                                        "/spectral_filter_TYPE[",
+                                        "/temp_control_TYPE[",
+                                        "/software_TYPE[",
+                                        "/LENS_OPT["
+
+        ]
+        if self.eln_data.get(key) is None:
+            for string in upper_and_lower_mixed_nexus_concepts:
+                key = key.replace(string,"/[")
+            # add only characters, if they are lower case and if they are not "[" or "]"
+            result = ''.join([char for char in key if not (char.isupper() or char in '[]')])
+            result = result.replace("entry","ENTRY[entry]") # CHANGE THIS LATER

```

Comment:
Here, I would make use of `self.callbacks.entry_name` like this
```suggestion
            result = result.replace("entry",f"ENTRY[{self.callbacks.entry_name}]")
```
 This can be useful if there are multiple entries in the future. You will have to test that it works like here though.

Reply:
okay, wasnt aware of that functionality. Sounds good and exactly what I needed to change later.